sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala

```diff
@@ -82,8 +96,30 @@ case class GpuIf(
     }
   }
 
-  override def columnarEval(batch: ColumnarBatch): Any = computeIfElse(batch, predicateExpr,
-    trueExpr, falseExpr.columnarEval(batch))
+  @transient
+  private[this] lazy val func = if (optAllTrueFalse) {
```

Comment:
Nit: `func` here is too generic, we can afford a better name for readability.  Actually `lazy val` can override a function def, so should be able to do something more direct like this:
```
override lazy val columnarEval(batch: ColumnarBatch): Any = if (...) {
```

Apples to the other places where this was done.

Reply:
I tried but got an error.
```
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala:222: not found: value columnarEval
[ERROR] one error found
```
Then, I changed to 
`override lazy val columnarEval: ColumnarBatch => Any = if (...) {`
, but still got another error
```
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala:222: lazy value columnarEval overrides nothing.
Note: the super classes of class GpuCaseWhen contain the following, non final members named columnarEval:
```
So I just renamed the function.

Reply:
I used `lazy val` to implement an abstrct method before.

Reply:
I decided to remove the config, so this is out of date.